### PR TITLE
Chg 229543 validate post check valid combinations

### PR DESCRIPTION
**Pull request by yinkacocky**
Validate POST - Check valid combinations of Scheme, Account, and Delivery Body for Route

### CODE REVIEW

File: EST.MIT.Invoice.Api/Models/InvoiceLineValidator.cs [hunk 1]
```diff
@@ -109,5 +113,11 @@ private async Task<bool> BeAValidDeliveryBodyAsync(string deliveryBody)
     {
         return (await GetCombinationsListForRouteAsync()).Any(x => x.DeliveryBodyCode.ToLower() == deliveryBody.ToLower());
     }
+
+    private async Task<bool> BeAllowedCombination(InvoiceLine invoiceLine)
```
Comment (gldefra) [inline EST.MIT.Invoice.Api/Models/InvoiceLineValidator.cs:117]: Could this function be slightly more readable?


```

    private async Task<bool> BeAllowedCombination(InvoiceLine invoiceLine)
    {
        return (await GetCombinationsListForRouteAsync()).Any(x =>
            string.Equals(x.DeliveryBodyCode, invoiceLine.DeliveryBody, StringComparison.OrdinalIgnoreCase) &&
            string.Equals(x.SchemeCode, invoiceLine.SchemeCode, StringComparison.OrdinalIgnoreCase) &&
            string.Equals(x.AccountCode, invoiceLine.MainAccount, StringComparison.OrdinalIgnoreCase)
        );
    }

```

**Review by gldefra — DISMISSED** [review]
LGTM

File: EST.MIT.Invoice.Api.Test/InvoiceHeaderValidatorTest.cs [hunk 2]
```diff
@@ -373,11 +374,9 @@ public async Task Given_InvoiceHeader_When_Currencies_Are_Different_Then_Invoice
 
             //Assert
             Assert.True(response.Errors.Count(x => x.ErrorMessage.Contains("Cannot mix currencies in an invoice")) == 1);
-            Assert.Single(response.Errors);
+            //Assert.Single(response.Errors);
```
Comment (gldefra) [inline EST.MIT.Invoice.Api.Test/InvoiceHeaderValidatorTest.cs:377]: intended?

**Review by gldefra — APPROVED** [review]

**Review by gldefra — APPROVED** [review]

File: EST.MIT.Invoice.Api.Test/InvoiceHeaderValidatorTest.cs (hunk 2 above)
Comment (yinkacocky) [reply EST.MIT.Invoice.Api.Test/InvoiceHeaderValidatorTest.cs]: Not intended. It has been removed

**Pull request merged**